compiler/rustc_error_messages/locales/en-US/lint.ftl

```diff
@@ -13,6 +13,8 @@ lint_enum_intrinsics_mem_variant =
     the return value of `mem::variant_count` is unspecified when called with a non-enum type
     .note = the type parameter of `variant_count` should be an enum, but it was instantiated with the type `{$ty_param}`, which is not an enum.
 
+lint_empty_for_iterator = empty for iterator
```

Comment:
I think this description and the name could be improved. Could the lint be renamed to `empty_iter_ranges`? The description could be like this:

```suggestion
lint_empty_iter_ranges = this `for` loop is never run
```